config/README.md

```diff
@@ -1,31 +1,62 @@
+The YAML files in this directory represent the [ko](https://github.com/google/ko) development
+level configuration use to work on the eventing-kafka source.  They will install eventing-kafka
+into the existing Kubernetes cluster which should have Knative Eventing installed.   It is also
+expected that the user will provide a viable Kafka cluster.
 
-## Kafka Providers
+## Usage
 
-Eventing-Kafka supports either a standard Kafka deployment or Azure EventHubs. In order to configure the desired provider,
-set the eventing-kafka.kafka.provider field in the [config-eventing-kafka configmap](200-eventing-kafka-configmap.yaml)
+As mentioned above these YAML files are to be installed with `ko`.  Thus, you need to set the
+`$KO_DOCKER_REPO` environment variable as desired.  If using `ko.local` then you'll need to
+change the `ImagePullPolicy` to `IfNotPresent`.
 
-The current allowed values are:
+The **data** values of the [kafka-secret.yaml](300-kafka-secret.yaml) should be correctly
+populated for the user provided Kafka cluster.  Similarly the **data** values of the
+[eventing-kafka-configmap.yaml](200-eventing-kafka-configmap.yaml) file should be configured
+for your particular use case.
 
-* `local`: Standard Kafka installation with no special authorization required (default)
-* `confluent`: Confluent Cloud
-* `azure`: Azure Event Hubs
+Install via `ko apply --strict -f ./config` from the parent/root directory in order to build
+and deploy the project.  The `--strict` option is really only needed when using the `custom`
+AdminType, but shouldn't hurt in other cases.
 
-The provider chosen effects how authentication as well as admin calls (topic creation, deletion etc) work.
+## Kafka Admin Types
+
+Eventing-Kafka supports a few options for the administration of Kafka Topics (Create / Delete) in the
+user provided Kafka cluster.  The desired mechanism is specified via the `eventing-kafka.kafka.adminType`
+field in [eventing-kafka-configmap.yaml](200-eventing-kafka-configmap.yaml) and must be one of `kafka`,
+`azure`, or `custom` as follows...
+
+- **kafka:** This is the normal / default use case that most users will want.  It uses the standard Kafka API (via the Sarama ClusterAdmin) for managing Kafka Topics in the cluster.
+- **azure:** Users of Azure EventHubs will know that Microsoft does not support the standard Kafka Topic administration and are required instead to use their API.  This option provides for such support via the Microsoft Azure EventHub Go client.
+- **custom:** This option provides an external hook for users who need to manage Topics themselves. This could be to support a proprietary Kafka implementation with a custom interface / API.  The user is responsible for implementing a sidecar Container with the expected HTTP endpoints.  They will need to add their sidecar Container to the [deployment.yaml](400-deployment.yaml).  Details for implementing such a solution can be found in the [Kafka README](../pkg/common/kafka/README.md).
```

Comment:
Yeah, because the stupid MD validation tool wouldn't take them otherwise - you can see my commit history (on the branch) for those lines - tried lots of different ways.